vos/vos/storage_inventory.py

```diff
@@ -0,0 +1,314 @@
+        stream = Stream(self.conn)
+        logger.debug('Uploading {}'.format(source))
+        return stream.upload(upload_url, destination, source, source_md5,
+                             self.get_metadata)
+
+    def _get_protocol(self):
+        if self.secure_get:
+            return 'https'
+        else:
+            return 'http'
+
+    def _is_remote(self, uri):
+        _uri = urlparse(uri)
+        return _uri.scheme and _uri.scheme != 'file'
+
+    def _get_ws_client(self, session_headers=None):
+        return net.BaseWsClient(self.resource_id, EndPoints.subject, VOS_AGENT,
+                                host=os.getenv('VOSPACE_WEBSERVICE', None),
+                                session_headers=session_headers)
+
+    def get_metadata(self, uri):
+        """
+        Obtain the general metadata for the artifact/node identified by the
+        given uri.
+
+        :param uri: An Artifact URI.
+        :type uri: unicode
+        :return: A dict with the following keys:
+                - content_disposition: unicode filename disposition
+                - content_encoding: unicode encoding value (i.e. gzip)
+                - content_length: long value; count of bytes
+                - content_md5: unicode hash value
+                - content_type: unicode content type string
+                                (i.e. application/fits)
+        :rtype: {}
+        """
+        logger.debug('vcp::get_metadata')
+        session_headers = {HEADER_DELEG_TOKEN: self.token}
+        ws_client = self._get_ws_client()
+        ws_client.session_headers = session_headers
+
+        response = ws_client.head(resource=(STANDARD_ID, uri))
+        headers = response.headers
+
+        return {
+            'content_disposition': os.path.split(uri)[-1],
+            'content_encoding': headers.get('Content-Encoding'),
+            'content_length': headers.get('Content-Length'),
+            'content_md5': headers.get('Content-MD5'),
+            'content_type': headers.get('Content-Type')
+        }
+
+    def isdir(self, uri):
+        """
+        Check to see if the given uri is or is a link to a containerNode.
+
+        :param uri: a URI string to test.
+        :rtype: bool
+        """
+        try:
+            _uri = urlparse(uri)
+            path = _uri.path
+
+            if not _uri.scheme or _uri.scheme == 'file':
+                return os.path.isdir(path) or path.endswith('/')
+            else:
+                return path.endswith('/')
+        except exceptions.NotFoundException:
+            return False
+
+    def isfile(self, uri):
+        """
+        Check if the given uri is or is a link to a DataNode
+
+        :param uri: the VOSpace Node URI to test.
+        :rtype: bool
+        """
+        try:
+            _uri = urlparse(uri)
+            path = _uri.path
+
+            if not _uri.scheme or _uri.scheme == 'file':
+                return os.path.isfile(path) or not path.endswith('/')
```

Comment:
You might also need to check if it's not a directory. `os.path.isdir('/tmp')` returns True